File: Clock In/EditNumberViewController.m
//
//  EditNumberViewController.m
//  Clock In
//
//  Created by Steve Ewell on 12/24/14.
//  Copyright (c) 2014 Steve Ewell. All rights reserved.
//

#import "EditNumberViewController.h"

@interface EditNumberViewController()

@property (weak, nonatomic) IBOutlet UITextField *editTextField;
@property (strong, nonatomic) NSString *employeeNumber;
@property (strong, nonatomic) UIToolbar *keyboardToolbar;
@property (strong, nonatomic) UIBarButtonItem *flexBarButton;
@property (strong, nonatomic) UIBarButtonItem *doneButton;

@end

@implementation EditNumberViewController

-(void)viewDidLoad {
    [super viewDidLoad];
    NSUserDefaults *defaults = [[NSUserDefaults alloc]initWithSuiteName:@"group.ewell.TodayExtensionSharingDefaults"];
    self.editTextField.text = [defaults stringForKey:@"employeeNumber"];
    self.keyboardToolbar = [[UIToolbar alloc]init];
    [self.keyboardToolbar sizeToFit];
    [self.editTextField becomeFirstResponder];
    self.flexBarButton = [[UIBarButtonItem alloc]initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:nil action:nil];
    self.doneButton = [[UIBarButtonItem alloc]initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self action:@selector(doneClicked:)];
    self.keyboardToolbar.items = [[NSArray alloc]initWithObjects:self.flexBarButton, self.doneButton, nil];
    self.editTextField.inputAccessoryView = self.keyboardToolbar;
    [self.editTextField becomeFirstResponder];
}

-(void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    NSUserDefaults *defaults = [[NSUserDefaults alloc]initWithSuiteName:@"group.ewell.TodayExtensionSharingDefaults"];
    self.employeeNumber = self.editTextField.text;
    if (self.employeeNumber.length > 12 || self.employeeNumber.length == 0) {
        [defaults synchronize];
    } else {
        [defaults setValue:self.employeeNumber forKey:@"employeeNumber"];
        [defaults synchronize];
    }
}

-(IBAction)doneClicked:(NSObject *)sender {
    NSUserDefaults *defaults = [[NSUserDefaults alloc]initWithSuiteName:@"group.ewell.TodayExtensionSharingDefaults"];
    self.employeeNumber = self.editTextField.text;
    if (self.employeeNumber.length > 12 || self.employeeNumber.length == 0) {
        UIAlertView *alert = [[UIAlertView alloc]initWithTitle:@"Number out of range!" message:@"Enter a number between 1 and 999999." delegate:self cancelButtonTitle:@"Ok" otherButtonTitles:nil];
        [alert show];
    } else {
        [defaults setValue:self.employeeNumber forKey:@"employeeNumber"];
        [defaults synchronize];
        [self performSegueWithIdentifier:@"toSettings" sender: self];
    }
    [self.view endEditing:YES];
}

- (void)alertView:(UIAlertView *)alertView didDismissWithButtonIndex:(NSInteger)buttonIndex {
    if (buttonIndex == 0) {
        NSUserDefaults *defaults = [[NSUserDefaults alloc]initWithSuiteName:@"group.ewell.TodayExtensionSharingDefaults"];
        [defaults setValue:@"123456" forKey:@"employeeNumber"];
        [defaults synchronize];
        [self.editTextField becomeFirstResponder];
    }
}

@end

File: Clock In/EditNumberViewController.swift
//
//  EditNumberViewController.swift
//  Clock In
//
//  Created by Steve Ewell on 11/27/14.
//  Copyright (c) 2014 Steve Ewell. All rights reserved.
//

import UIKit

class EditNumberViewController: UITableViewController {

    @IBOutlet weak var numberEditableCell: SettingsEditableCell!
    
    var keyboardToolbar:UIToolbar = UIToolbar()
    var flexBarButton:UIBarButtonItem = UIBarButtonItem()
    var doneButton:UIBarButtonItem = UIBarButtonItem()
    
    
    var employeeNumber:String = String()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        employeeNumber = NSUserDefaults.standardUserDefaults().stringForKey("employeeNumber")!
        numberEditableCell.cellTextField.text = employeeNumber
        
        keyboardToolbar.sizeToFit()
        
        flexBarButton = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.FlexibleSpace, target: nil, action: nil)
        
        doneButton = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.Done, target: self, action: "doneClicked:")
        
        keyboardToolbar.items = [flexBarButton, doneButton]
        self.numberEditableCell.cellTextField.inputAccessoryView = keyboardToolbar
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func doneClicked(AnyObject) {
        employeeNumber = numberEditableCell.cellTextField.text
        if (employeeNumber.length() > 12 || employeeNumber.length() == 0) {
            let alert = UIAlertController(title: "Number out of range!", message: "Enter a number between 1 and 999999.", preferredStyle: .Alert)
            alert.addAction(UIAlertAction(title: "Ok", style: .Default, handler: alertHandler))
            self.presentViewController(alert, animated: true, completion: nil)
        } else {
            NSUserDefaults.standardUserDefaults().setValue(employeeNumber, forKey: "employeeNumber")
            NSUserDefaults.standardUserDefaults().synchronize()
        }
        self.view.endEditing(true)
    }
    
    func alertHandler(act:UIAlertAction!) {
        NSUserDefaults.standardUserDefaults().setValue("123456", forKey: "employeeNumber")
        NSUserDefaults.standardUserDefaults().synchronize()
    }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using [segue destinationViewController].
        // Pass the selected object to the new view controller.
    }
    */

}
